src/Ansatz/Chain.jl

```diff
@@ -343,6 +343,51 @@ function isrightcanonical(qtn::Chain, site; atol::Real = 1e-12)
     return isapprox(contracted, identity_matrix; atol)
 end
 
+canonize(tn::Chain, args...; kwargs...) = canonize!(deepcopy(tn), args...; kwargs...)
+canonize!(tn::Chain, args...; kwargs...) = canonize!(boundary(tn), tn, args...; kwargs...)
+
+"""
+canonize(boundary::Boundary, tn::Chain)
+
+Transform a `Chain` tensor network into the canonical form (Vidal form), that is,
+we have the singular values matrix Λᵢ between each tensor Γᵢ₋₁ and Γᵢ.
+"""
+function canonize!(::Open, tn::Chain)
+    S_vals = Tensor[]
+
+    # left-to-right QR sweep, get left-canonical tensors
+    for i in 1:nsites(tn)-1
+        canonize_site!(tn, Site(i); direction = :right, method = :qr)
+    end
+
+    # right-to-left SVD sweep, get right-canonical tensors
+    for i in nsites(tn):-1:2
+        canonize_site!(tn, Site(i); direction = :left, method = :svd)
+
+        # extract the singular values and contract them with the next tensor
+        Λᵢ₋₁ = pop!(TensorNetwork(tn), select(tn, :between, Site(i - 1), Site(i)))
+        Bᵢ₋₁ = select(tn, :tensor, Site(i - 1))
+        replace!(TensorNetwork(tn), Bᵢ₋₁ => contract(Bᵢ₋₁, Λᵢ₋₁, dims = ()))
+        push!(S_vals, Λᵢ₋₁)
+    end
+    reverse!(S_vals) # reverse the singular values to match the order of the tensors
```

Comment:
You can change the order:
1. right-to-left QR sweep
2. left-to-right SVD sweep

This way you don't need the `reverse!`